### ISSUE TITLE
Add ability to copy stream URL to clipboard

### ISSUE THREAD

**Issue by Beep6581** [clementine-player/Clementine#4774]
I have a bunch of radio streams in a playlist, and it would be nice to be able to right-click on one and to copy its URL so I can share it. As is, its not only impossible to find the actual URL, but its even impossible to figure out where a stream in the playlist came from (Icecast? SKY.fm? SomaFM? other?) without manually searching for it. Again, this would be a great addition for a right-click context menu.

Please note, the URL should be the one directly to the stream, not some general one. E.g. right-click on "Cafe de Paris" in Internet > SKY.fm, and the closest option to getting a URL is "Open www.sky.fm in browser" but that's a dead end since you then have to search for that station again on their website and their website requires Adobe Flash to play a stream, which more and more people refuse to use for safety and other reasons.

**Comment by jlsjonas** [comment]
+1 for adding copy URL support for services like soundcloud

**Comment by coconutlulz** [comment]
I've been looking into this request. You can add the 'filename' column to your playlist display and this will show you both the URL and the service. Copying the URL might not help as some of them (e.g. Soundcloud) require the Clementine API key. While this would be copied, too, I'm not sure what restrictions there are on the Soundcloud API in terms of sharing it to be used from other applications.

If you are unhappy with the above workaround, I'll attempt to develop the feature and open a pull request.

**Comment by Beep6581** [comment]
Thank you for looking into this.

I couldn't see the stream URLs, but found that when I replace "File name (without path)" with "File name" then they appear. This is good to know, but it would be far more practical to be able to right-click an entry and "Copy path" to clipboard, ending up with `/music/Mike Oldfield/2003 - Tubular Bells 2003/2 - Fast Guitars.ogg` or `http://streaming203.radionomy.com:80/OrabidooblogRadio` or whatever.

**Comment by k3dar** [comment]
i too vote... please add "Copy Path/URL" to song context menu, or maybe "Show Information" for showing window with informations about song and with selectable&copyable text... thx :)

**Comment by suther** [comment]
+10

**Comment by jbauer180266** [comment]
On a related note: For Digitally Imported you show di:// domains, which aren't too useful. Replacing this prefix with http://pub7.di.fm/di_ and appending the name makes those URLs valid and much more useful, e.g. http://pub7.di.fm/di_deeptech [no warranty for linked page]

[Also, let me say that I'm blown away by Clementine - what an amazing player. And even the "less serious" features like Rain or the Nyanalayzer cat simply rock. I hope this player stays around for a long while, because - at least in my opinion - it really blows the competition out of the water. Great work.]

**Comment by coconutlulz** [comment]
Hi everyone,

Just so you know, I'm working on this on my own fork at the moment. So far, I have completed Soundcloud URL copying. It might be a couple of days before I'm ready to open a pull request.

**Comment by Beep6581** [comment]
Great!

**Comment by coconutlulz** [comment]
Soundcloud and SomaFM working. I hope to have everything finished by the end of the week.

**Comment by coconutlulz** [comment]
Hey all. The pull request is open. I didn't forget about you! :dancers:

**Comment by doublerebel** [comment]
+1 LGTM

**Comment by jlsjonas** [comment]
If only someone at the team would actually look at their issue & pull request list *sighs*

**Comment by Beep6581** [comment]
@hatstand

**Comment by Ferroin** [comment]
@jlsjonas You do realize that nobody who works on Clementine is paid to do so, does so entirely in their free time, and they have no obligation whatsoever to you or anyone else here...

**Comment by Beep6581** [comment]
@Ferroin you do realize that that is irrelevant?

**Comment by jlsjonas** [comment]
@Ferroin yes, I do. I'm just saying that if (out-of-team) community effort is done it would be nice if the team would actually save themselves time by (relatively quickly) verifying code-quality & accepting the pull request so that everyone (including themselves) can benefit from it.
Especially if you compare the average merge time ;)

it's just a shame that github marks this as an issue instead of a pull request now that it actually is a pull request

**Comment by affinityv** [comment]
Podcasts... it would be good to be able to edit/see/copy/export the URL(s) of rss feeds.

A workaround to view the details is to run sqlite3 and query the podcasts table; but this should be able to be done in the UI.

Also, fwiw, the "Edit preferences" dialog.... it would be better if the "Apply" button was greyed out unless a change has been made.  But this is out of scope on this particular issue, even though I thought that perhaps I could see the URL(s) in preferences somehow as it wasn't available elsewhere.

**Comment by Akronix** [comment]
This is specially important for SoundCloud because [they state the attributions you have to meet to stream sound from their API.](https://developers.soundcloud.com/docs/api/guide#playing)
And there, they explicitly say:
>     3. Link to the SoundCloud URL containing the work
>     4. If the sound is private link to the profile of the creator
